From 3bf0bbdc8fd511b8c9d77700daf54869e57f0202 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Fri, 7 Sep 2018 15:32:18 +0800
Subject: [PATCH 4564/5242] MLK-19493 gpu: imx: dpu: tcon: Tune kachuck from
 slave

commit  d85179a0c7422b38a33b0a4703d21678b142d234 from
https://source.codeaurora.org/external/imx/linux-imx.git

As suggested by designer for safety's sake, this patch tunes
kachuck from slave tcon to generate it 2 lines later than
master tcon.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-tcon.c |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/imx/dpu/dpu-tcon.c b/drivers/gpu/imx/dpu/dpu-tcon.c
index 60b7a57..a4ba1e7 100644
--- a/drivers/gpu/imx/dpu/dpu-tcon.c
+++ b/drivers/gpu/imx/dpu/dpu-tcon.c
@@ -197,10 +197,13 @@ void tcon_cfg_videomode(struct dpu_tcon *tcon,
 	y = m->vdisplay;
 	/*
 	 * If sync mode fixup is present, the kachuck signal from slave tcon
-	 * should be one line later than the one from master tcon.
+	 * should be two lines later than the one from master tcon.
 	 */
-	if (side_by_side && tcon_is_slave(tcon) && devtype->has_syncmode_fixup)
+	if (side_by_side && tcon_is_slave(tcon) &&
+	    devtype->has_syncmode_fixup) {
 		y++;
+		y++;
+	}
 	dpu_tcon_write(tcon, X(0xa) | Y(y), SPGPOSON(4));
 	dpu_tcon_write(tcon, 0x0, SPGMASKON(4));
 
-- 
1.7.9.5

